### PR TITLE
added list[Levels] to Dataframe.stack()

### LINKED ISSUE #489: DataFrame.stack() needs to accept list for level argument
**Describe the bug**

For `DataFrame.stack()`, `level` is allowed to be a list

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
```python
import pandas as pd
multicol2 = pd.MultiIndex.from_tuples([('weight', 'kg'),
                                       ('height', 'm')])
df_multi_level_cols2 = pd.DataFrame([[1.0, 2.0], [3.0, 4.0]],
                                    index=['cat', 'dog'],
                                    columns=multicol2)
df_multi_level_cols2.stack([0, 1])
```

2. Indicate which type checker you are using (`mypy` or  `pyright`).

pyright 1.1.286

3. Show the error message received from that type checker while checking your example.
pyright reports:
```text
stack.py:7:28 - error: Argument of type "list[int]" cannot be assigned to parameter "level" of type "Level" in function "stack"
    Type "list[int]" cannot be assigned to type "Level"
      "list[int]" is incompatible with protocol "Hashable"
        "__hash__" is an incompatible type
          Type "None" cannot be assigned to type "(self: list[int]) -> int"
      "list[int]" is incompatible with "int" (reportGeneralTypeIssues)
```

**Please complete the following information:**
 - OS: Windows 10
 - OS Version:  10
 - python version: 3.9
 - version of type checker:  pyright 1.1.286
 - version of installed `pandas-stubs`   1.5.2.221213

**Additional context**

Change
https://github.com/pandas-dev/pandas-stubs/blob/ab1de1f65e896dfc0cc67feb55ffa61a715abdd7/pandas-stubs/core/frame.pyi#L1000
to allow `Level | list[Level]` for the `level` argument.

PR welcome with tests

### PR DESCRIPTION
**Pull request by muddi900**
- [x] Closes #489 
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Can you please add a test corresponding to the code from the issue, and also an example that tests a single level?

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: tests/test_frame.py [hunk 1]
```diff
@@ -1798,6 +1798,25 @@ def test_frame_index_numpy() -> None:
...
+        [[1.0, 2.0], [3.0, 4.0]], index=["cat", "dog"], columns=multicol2
+    )
+
+    check(
+        assert_type(df_multi_level_cols2.stack(0), Union[pd.DataFrame, pd.Series[Any]]),
+        pd.DataFrame,
+    )
+    check(
+        assert_type(
+            df_multi_level_cols2.stack([0, 1]), Union[pd.DataFrame, pd.Series[Any]]
```
[the remark this answers is not in the document]
Comment (muddi900) [reply tests/test_frame.py:1814]: When I don't use the generic it fails pyright tests on 3.10.

```bash
 /home/muddi900/src/pandas-stubs/tests/test_frame.py:1809:21 - error: "assert_type" mismatch: expected "DataFrame | Series[Unknown]" but received "DataFrame | Series[Any]" (reportGeneralTypeIssues)
  /home/muddi900/src/pandas-stubs/tests/test_frame.py:1813:21 - error: "assert_type" mismatch: expected "DataFrame | Series[Unknown]" but received "DataFrame | Series[Any]" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 informations 
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Based on this: https://github.com/pandas-dev/pandas-stubs/actions/runs/3831371047/jobs/6520515609

you need to run the pre-commit action to format the files.

If you do `poetry run poe style`, that should fix the issue.

**Comment by Dr-Irv** [comment]
@muddi900 It turns out that we had an issue with our CI due to a SQLAlchemy update.  This was just fixed in #496. 

Can you merge with upstream/main and push, and ping here so I can trigger the tests?  Code looks good at this point, so just need it to pass our CI.

**Review by Dr-Irv — APPROVED** [review]
Thanks @muddi900

**Pull request merged**